app/helpers/sessions_helper.rb

```diff
@@ -0,0 +1,24 @@
+module SessionsHelper
+  def log_out
+    session.delete :user_id
+    @current_user = nil
+  end
+
+  def log_in user
+    session[:user_id] = user.id
+  end
+
+  def current_user
+    @current_user ||= User.find_by id: session[:user_id]
```

Comment:
Giải thích tại sao chỗ này dùng toán tử ||= làm gì?
Mà không gán = luôn ?

Reply:
Cái này là em kiểm tra xem có thằng current _user hay không nếu nil thì sẽ gán nó vs kết quả câu truy vấn. Còn tại sao ko gán luôn nếu nó có thỉ đở truy vấn đỡ tốn tài nguyên